### ISSUE TITLE
error "Generate features from stapled peptide sequence"

### ISSUE THREAD

**Issue by baba-hashimoto** [dahuilangda/stapep_package#1]
Thank you for making this wonderful program available to the public.”
I ran the program “Generate features from stapled peptide sequence” and found a bug, so I contacted you. The bug is as follows.
length: 19
weight: 2413.9530000000004
hydrophobicity index: -0.2000000000000001
charge 5.996
charge_density 0.0024838926027143026
aromaticity 0.05263157894736842
fraction_arginine 0.2631578947368421
fraction_lysine 0.05263157894736842
Traceback (most recent call last):
File “/home/stapep_package/stapep/example/test3.py”, line 15, in <module>
print('lyticity index: ', pps.lyticity_index) # lyticity index (Mourtada and Rida, 2019)
File “/myfolderpath/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/utils.py”, line 194, in lyticity_index
raise ValueError(f'{aa} is not a valid amino acid.')
ValueError: B is not a valid amino acid.

I would be grateful if you could tell me how to solve this.

**Comment by dahuilangda** [comment]
Hi @baba-hashimoto 

Thank you for your feedback. I have fixed the bug related to ValueError: B is not a valid amino acid. and have pushed the updated code to the repository. Please pull the latest changes to your local setup to apply this fix. If you encounter any issues or need further assistance, please feel free to reach out.

**Comment by baba-hashimoto** [comment]
@dahuilangda 
Thank you for the correction.
I ran the following program.

from stapep.utils import ProtParamsSeq

seq = 'Ac-BATP-R8-RRR-Aib-BLBR-R3-FKRLQ' # Define the peptide sequence
pps = ProtParamsSeq(seq) # Initialize the class

-Get the features
print('length: ', pps.seq_length) # length of the peptide sequence (without N-terminal and C-terminal)
print('weight: ', pps.weight) # molecular weight
print('hydrophobicity index: ', pps.hydrophobicity_index) # hydrophobicity index (Kyte and Doolittle, 1982)
print('charge', pps.calc_charge(pH=7.0)) # charge at pH=7.0
print('charge_density', pps.calc_charge_density(pH=7.0)) # charge density at pH=7.0 (charge/length)
print('aromaticity', pps.aromaticity) # aromaticity (number of aromatic amino acids/length)
print('fraction_arginine', pps.fraction_arginine) # fraction of arginine (number of arginine/length)
print('fraction_lysine', pps.fraction_lysine) # fraction of lysine (number of lysine/length)
print('lyticity index: ', pps.lyticity_index) # lyticity index (Mourtada and Rida, 2019)
print('isoelectric_point: ', pps.isoelectric_point) # isoelectric point

 -Output:
#length: 19
#weight: 2413.9530000000004
#hydrophobicity index: -0.2000000000000001
#charge 5.996
#charge_density 0.0024838926027143026
#aromaticity 0.05263157894736842
#fraction_arginine 0.2631578947368421
#fraction_lysine 0.05263157894736842
#lyticity index: 114.16199999999999
#isoelectric_point: 11.999967765808105
l

My result
ength: 19
weight: 2413.9530000000004
hydrophobicity index: -0.2000000000000001
charge 5.996
charge_density 0.0024838926027143026
aromaticity 0.05263157894736842
fraction_arginine 0.2631578947368421
fraction_lysine 0.05263157894736842
lyticity index: 564.745
isoelectric_point: 11.999967765808105

Only the lysis index shows a different value. If there is any mistake, I would be grateful if you could comment on it.

**Comment by dahuilangda** [comment]
@baba-hashimoto  Thank you for sharing your results. It seems that your calculation for the lyticity index is indeed correct. The error in the README regarding the lyticity index was due to an incorrect calculation formula in our previous code. Once we identified this issue, we revised the code to implement the correct formula, ensuring that the lyticity index is now accurately calculated. I have updated the README to reflect these corrections. Thank you for your help in catching this mistake and for your understanding as we strive to improve the accuracy of our documentation.
